lib/actions/actionsBuilder.js

```diff
@@ -0,0 +1,30 @@
+const AddLinkToProjectAction = require('./AddLinkToProjectAction');
+const assert = require('assert');
+
+class ActionsBuilder {
+  constructor(){
+    this.commandMap = {
+      'add': new AddLinkToProjectAction,
+    }
+  }
+
+  parseActionTerm(userInputString = '') {
+    assert(typeof userInputString === 'string');
+
+    let parsedUserInput = userInputString.toLowerCase().split(' ');
+
+    if (parsedUserInput.length < 3){
```

Comment:
This is an interesting requirement. Why do we care about this?

Reply:
I want to fail fast if the input isn't correct.
assert checks for dev errors not user errors --> if the input is not a string it immediately blows up
we can assume that the command has 3 arguments add param param. When we parse this input, we should expect parsedUserInput to have a length of 3. If it is less, the users input is invalid.

Reply:
I don't think this method cares about anything other than the first term in the message text. This is a wonderful example of a leaky abstraction. By hard coding the structure of message text beyond the part influencing this classes responsibility, you hinder the ability of the message text structure to change in the future. Limit the concerns of this class to _just the first term_.

Reply:
What are some thing that I should check for? 
The concerns of the first term is that it follows the command map with the terms 'add', 'delete' etc... should I be concerned with this at this very moment or no?

Reply:
You just care if it results in an `Action` instance. So take that first term, look up the action. If you get one 🙌. If not, throw. Simple. No assumptions about the future, etc. If you get no term, you might make an explicit throw of "no term found" for easier diagnostics later.

Reply:
this makes a lot of sense here ^